ngraph/core/src/op/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,175 @@
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v6::CTCGreedyDecoderSeqLen::CTCGreedyDecoderSeqLen(const Output<Node>& input,
+                                                       const Output<Node>& seq_len,
+                                                       const Output<Node>& blank_index,
+                                                       const bool merge_repeated,
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len, blank_index})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::CTCGreedyDecoderSeqLen::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v6_CTCGreedyDecoderSeqLen_validate_and_infer_types);
+    const auto& logits_pshape = get_input_partial_shape(0);
+    const auto& seq_len_pshape = get_input_partial_shape(1);
+    auto input_et = get_input_element_type(0);
+    const bool logits_is_static_rank = logits_pshape.rank().is_static();
+    const bool seq_len_is_static_rank = seq_len_pshape.rank().is_static();
+
+    // output dynamic rank tensor if all inputs are of dynamic rank
+    if (!logits_is_static_rank && !seq_len_is_static_rank)
+    {
+        set_output_type(0, input_et, PartialShape{Dimension::dynamic(), Dimension::dynamic()});
+    }
+
+    // check ranks of input tensors
+    if (logits_is_static_rank)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              logits_pshape.rank().get_length() == 3,
+                              "The rank of logits tensor must be equal to 3.");
+    }
+    if (seq_len_is_static_rank)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              seq_len_pshape.rank().get_length() == 1,
+                              "The rank of sequence len tensor must be equal to 1.");
+    }
+
+    // check optional input type: blank index
+    if (get_input_size() == 3)
+    {
+        const auto& blank_index_type = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              blank_index_type.is_integral_number(),
+                              "The blank index type is expected to be an integer type. Got: ",
+                              blank_index_type);
+
+        const auto& blank_index_shape = get_input_partial_shape(2);
+        Shape blank_index_type_shape = blank_index_shape.to_shape();
```

Comment:
The `to_shape` method can be called for a static shape only. Please, add a check that `blank_index_shape` is static.

Reply:
Also, add a unit test for this case.

Reply:
Also, the variable should be called `blank_index_shape` instead of `blank_index_type_shape`